### ISSUE TITLE
CUDA CODE not compiling

### ISSUE THREAD

**Issue by AssassinCroc** [zhanghang1989/PyTorch-Encoding#66]
I dont think the current code compiles the cuda code if you git clone and just do a `python setup.py install`. I inserted rubbish in cu file and followed your steps..no errors.. Cant even see any nvcc in the code to compile the code.

**Comment by zhanghang1989** [comment]
It uses the JIT https://pytorch.org/docs/master/cpp_extension.html?highlight=cppextension#torch.utils.cpp_extension.CppExtension

**Comment by zhanghang1989** [comment]
Have you installed the recent PyTorch version?

**Comment by AssassinCroc** [comment]
Do you mean 0.5.0 or 0.4.0 ? I am using pytorch  from nvcr.io/nvidia/pytorch docker repo.

**Comment by zhanghang1989** [comment]
I think your version is slightly behind the pytorch master branch, because `AT_ERROR` api recently has an update. You can either remove ROI Align from the code or update the PyTorch version.

**Comment by Yozey** [comment]
Same issue with `AT_ASSERT`, which makes encoding kernel, sync batch normalization and ROI align modules not compatible with the 0.4.0 pytorch version

**Comment by zhanghang1989** [comment]
This code is not compatible with pytorch 0.4.0. Please use the `master` branch and compile from source, or wait for next stable release. Thanks!

**Comment by baiyancheng20** [comment]
Have you solved the problem? @Yozey @AssassinCroc @zhanghang1989 
I use the pytorch 0.4.0 and I also met with this problem.

**Comment by Yozey** [comment]
I went back to use the previous version of this code, which is compatible with pytorch 0.3.0. I don't know if the previous version is still available online.

**Comment by zhanghang1989** [comment]
Please compile PyTorch from source on `master` branch of github.

**Comment by karenyun** [comment]
Hi, if I use the pytorch 0.4.0 and this master branch, how can I remove the incompatible part to let the "import encoding" work?

**Comment by zhanghang1989** [comment]
Since PyTorch 0.4 doesn't have JIT. You can manually install `enclib_cpu` and `enclib_gpu` by running 

https://github.com/zhanghang1989/PyTorch-Encoding/blob/master/encoding/lib/cpu/setup.py
https://github.com/zhanghang1989/PyTorch-Encoding/blob/master/encoding/lib/gpu/setup.py

Then change https://github.com/zhanghang1989/PyTorch-Encoding/blob/master/encoding/lib/__init__.py

**Comment by huangxGo** [comment]
Could you please tell me how to change the file?  [](https://github.com/zhanghang1989/PyTorch-Encoding/blob/master/encoding/lib/__init__.py)
THANKS!

**Comment by zhuizhunew** [comment]
activation_kernel.cu(21):error:__host__ or __device__ annotation on lambda requires --expt-extended-lambda nvcc flag

what is the problem?
